app/views/govuk_publishing_components/components/_single_page_notification_button.html.erb

```diff
@@ -0,0 +1,23 @@
+<%
+  page ||= ''
+  data_attributes ||= {}
+  base_path ||= nil
+  local_assigns[:margin_bottom] ||= 3
+  already_subscribed ||= false
+  text ||= already_subscribed ? t('components.single_page_notification_button.unsubscribe_text') : t('components.single_page_notification_button.subscribe_text')
+
+  shared_helper = GovukPublishingComponents::Presenters::SharedHelper.new(local_assigns)
+  wrapper_classes = %w(gem-c-single-page-notification-button govuk-!-display-none-print)
+  wrapper_classes << shared_helper.get_margin_bottom
+  classes = "govuk-body-s gem-c-single-page-notification-button__submit"
+%>
+<% button_text = capture do %>
+  <svg class="gem-c-single-page-notification-button__icon" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" height="18" width="18" viewBox="0 0 459.334 459.334"><path fill="currentColor" d="M177.216 404.514c-.001.12-.009.239-.009.359 0 30.078 24.383 54.461 54.461 54.461s54.461-24.383 54.461-54.461c0-.12-.008-.239-.009-.359H175.216zM403.549 336.438l-49.015-72.002v-89.83c0-60.581-43.144-111.079-100.381-122.459V24.485C254.152 10.963 243.19 0 229.667 0s-24.485 10.963-24.485 24.485v27.663c-57.237 11.381-100.381 61.879-100.381 122.459v89.83l-49.015 72.002a24.76 24.76 0 0 0 20.468 38.693H383.08a24.761 24.761 0 0 0 20.469-38.694z"/></svg><%= text %>
+<% end %>
+<%= tag.form class: wrapper_classes, action: "/email/subscriptions/single-page/new", method: "POST", data: data_attributes do %>
```

Comment:
Just want to check that putting the data attributes on the form and not the button is what is needed here. It might be, I don't know, just thought I'd check.

Reply:
Yep, in this instance I think they're needed on the form